From 7a5aadc76d5506a203abac07466682e9e5cf54e9 Mon Sep 17 00:00:00 2001
From: Zidan Wang <zidan.wang@freescale.com>
Date: Fri, 27 Nov 2015 13:42:41 +0800
Subject: [PATCH 0706/5242] MLK-11915-10 ASoC: imx_mqs: fix mqs_priv->name
 overrun issue

commit  085303d6fa6e296ae1904f3570db0751cf267650 from
https://source.codeaurora.org/external/imx/linux-imx.git

fix mqs_priv->name overrun issue. Reported by Coverity.

Signed-off-by: Zidan Wang <zidan.wang@freescale.com>
(cherry picked from commit ef1c59f93981b15412a0207d5517a26901bb2ecd)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/codecs/fsl_mqs.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/codecs/fsl_mqs.c b/sound/soc/codecs/fsl_mqs.c
index 2102391..3252230 100644
--- a/sound/soc/codecs/fsl_mqs.c
+++ b/sound/soc/codecs/fsl_mqs.c
@@ -165,7 +165,7 @@ static int fsl_mqs_probe(struct platform_device *pdev)
 		return -ENOMEM;
 
 	mqs_priv->pdev = pdev;
-	strcpy(mqs_priv->name, np->name);
+	strncpy(mqs_priv->name, np->name, sizeof(mqs_priv->name) - 1);
 
 	gpr_np = of_parse_phandle(np, "gpr", 0);
 	if (IS_ERR(gpr_np)) {
-- 
1.7.9.5

